Microsoft.Toolkit.Uwp.UI/Extensions/TextBox/TextBoxExtensions.Regex.Internals.cs

```diff
@@ -70,6 +70,12 @@ private static void ValidateTextBox(TextBox textBox, bool force = true)
                 case ValidationType.Decimal:
                     regexMatch = textBox.Text.IsDecimal();
                     break;
+                case ValidationType.Currency:
+                    // @"(^\d*\.\d{2}$)" regex pattern to detect currency sign with currency value
+                    // Mathes: $100.00, $100, $10.25
+                    // Non-Matches: 100., $10.233, $10.
+                    regexMatch = Regex.IsMatch(textBox.Text, @"^\$( )*\d*(.\d{1,2})?$");
```

Comment:
Until we can reference [this new extension method](https://github.com/CommunityToolkit/dotnet/pull/73), let's use the .NET Api directly so we can close this off :)

```suggestion
                    // Parse the entered data into a decimal, allowing existing currency symbols. Uses current UI culture.
                    regexMatch = decimal.TryParse(value, NumberStyles.Currency, out _);
```

Reply:
The `TryParse` method doesn't accept 3 parameters, there needs to be an `IFormatProvider` parameter as well.

I've written the code for this with `CultureInfo.CurrentCulture` as the `IFormatProvider` like you did in the PR for the new extension method (unless @olegL1337 is still planning on updating this PR)

My only question is, it this the best way to validate the currencies? with the `IFormatProvider` set to `CultureInfo.CurrentCulture` the validation will only pass for the user's local currency. Is there a way of including all currency cultures so that any currency will pass as valid?

Reply:
Afaik, there isn't a `CultureInfo` that represents "all" cultures. However, we can get all available cultures and check against each one separately.

```csharp
static decimal? TryParseCurrency(string str)
{
	foreach (var culture in CultureInfo.GetCultures(CultureTypes.AllCultures))
	{
		if (decimal.TryParse(str, NumberStyles.Currency, culture, out var amount))
				return amount;
	}

	return null;
}
```

See it in action: https://dotnetfiddle.net/HW8wZ4

Reply:
This feels this needs more options.

Perhaps we add a way to provide the current or a specific culture for `ValidationType.Currency`, and add an additional `ValidationType.AnyCurrency` if they'd like to check all cultures. @michael-hawker what do you think?